### DISCUSSION TITLE
2023-09-11 백엔드 정기 회의

### DISCUSSION THREAD

**Discussion by scarf005** [jiphyeonjeon-42/backend#765] [General]
안건을 자유롭게 추가해주세요.

**Comment by scarf005** [comment]
# 프론트엔드와 V2 연동

대회가 끝나 적용을 진행해도 될 것 같습니다. 현재 v2 reviews API까지만 연동 진행되었습니다.

https://github.com/jiphyeonjeon-42/backend/discussions/689#discussioncomment-6830193

**Comment by scarf005** [comment]
# V2 API 경로 설계하기

새롭게 작성하는 것이므로, V2 API의 경로와 입출력을 재설계하면 다음과 같은 장점이 있을 것 같습니다

## 일관성 개선

예시:

### GET `/api/books/search` -> GET `/books/{id}`
### POST `/api/books/create` -> POST `/books/{id}`와 POST `/info/{id}`로 분리

- REST API 준수 (동사 제거)
- 단일 책임 원칙 준수

### PATCH `/api/reservations/cancel/{id}` -> DELETE `/api/reservations/{id}`

## API 사용 편의 개선

예시: 도서 상태 반환시 숫자 대신 "ok", "lost", "damaged" 반환

**Reply by nyj001012** [reply]
이번주 수요일 14시에 API 경로 회의
